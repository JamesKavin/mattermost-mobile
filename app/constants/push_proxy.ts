// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const PUSH_PROXY_STATUS_VERIFIED = 'verified';
export const PUSH_PROXY_STATUS_UNKNOWN = 'unknown';
export const PUSH_PROXY_STATUS_NOT_AVAILABLE = 'not_available';

export const PUSH_PROXY_RESPONSE_VERIFIED = 'true';
export const PUSH_PROXY_RESPONSE_UNKNOWN = 'unknown';
export const PUSH_PROXY_RESPONSE_NOT_AVAILABLE = 'false';

export default {
    PUSH_PROXY_STATUS_VERIFIED,
    PUSH_PROXY_STATUS_UNKNOWN,
    PUSH_PROXY_STATUS_NOT_AVAILABLE,
    PUSH_PROXY_RESPONSE_VERIFIED,
    PUSH_PROXY_RESPONSE_UNKNOWN,
    PUSH_PROXY_RESPONSE_NOT_AVAILABLE,
};
