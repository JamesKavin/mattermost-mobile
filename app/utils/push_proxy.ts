// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {IntlShape} from 'react-intl';
import {Alert} from 'react-native';

import {PUSH_PROXY_RESPONSE_NOT_AVAILABLE, PUSH_PROXY_RESPONSE_UNKNOWN, PUSH_PROXY_STATUS_NOT_AVAILABLE, PUSH_PROXY_STATUS_UNKNOWN, PUSH_PROXY_STATUS_VERIFIED} from '@constants/push_proxy';
import EphemeralStore from '@store/ephemeral_store';

export function canReceiveNotifications(serverUrl: string, verification: string, intl: IntlShape) {
    switch (verification) {
        case PUSH_PROXY_RESPONSE_NOT_AVAILABLE:
            EphemeralStore.setPushProxyVerificationState(serverUrl, PUSH_PROXY_STATUS_NOT_AVAILABLE);
            alertPushProxyError(intl);
            break;
        case PUSH_PROXY_RESPONSE_UNKNOWN:
            EphemeralStore.setPushProxyVerificationState(serverUrl, PUSH_PROXY_STATUS_UNKNOWN);
            alertPushProxyUnknown(intl);
            break;
        default:
            EphemeralStore.setPushProxyVerificationState(serverUrl, PUSH_PROXY_STATUS_VERIFIED);
    }
}

export function alertPushProxyError(intl: IntlShape) {
    Alert.alert(
        intl.formatMessage({
            id: 'alert.push_proxy_error.title',
            defaultMessage: 'Notifications cannot be received from this server',
        }),
        intl.formatMessage({
            id: 'alert.push_proxy_error.description',
            defaultMessage: 'Due to the configuration for this server, notifications cannot be received in the mobile app. Contact your system admin for more information.',
        }),
        [{
            text: intl.formatMessage({id: 'alert.push_proxy.button', defaultMessage: 'Okay'}),
        }],
    );
}

export function alertPushProxyUnknown(intl: IntlShape) {
    Alert.alert(
        intl.formatMessage({
            id: 'alert.push_proxy_unknown.title',
            defaultMessage: 'Notifications could not be received from this server',
        }),
        intl.formatMessage({
            id: 'alert.push_proxy_unknown.description',
            defaultMessage: 'This server was unable to receive push notifications for an unknown reason. This will be attempted again next time you connect.',
        }),
        [{
            text: intl.formatMessage({id: 'alert.push_proxy.button', defaultMessage: 'Okay'}),
        }],
    );
}
