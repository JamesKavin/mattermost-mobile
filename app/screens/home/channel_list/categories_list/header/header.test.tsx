// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import {PUSH_PROXY_STATUS_VERIFIED} from '@constants/push_proxy';
import {renderWithIntl} from '@test/intl-test-helper';

import Header from './header';

describe('components/channel_list/header', () => {
    it('Channel List Header Component should match snapshot', () => {
        const {toJSON} = renderWithIntl(
            <Header
                pushProxyStatus={PUSH_PROXY_STATUS_VERIFIED}
                canCreateChannels={true}
                canJoinChannels={true}
                canInvitePeople={true}
                displayName={'Test!'}
            />,
        );

        expect(toJSON()).toMatchSnapshot();
    });
});
