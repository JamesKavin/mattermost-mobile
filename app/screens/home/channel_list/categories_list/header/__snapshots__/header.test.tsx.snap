// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/channel_list/header Channel List Header Component should match snapshot 1`] = `
<View
  animatedStyle={
    {
      "value": {
        "marginLeft": 0,
      },
    }
  }
  collapsable={false}
  style={
    {
      "marginLeft": 0,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontFamily": "Metropolis-SemiBold",
            "fontSize": 28,
            "fontWeight": "600",
            "lineHeight": 36,
          }
        }
        testID="channel_list_header.team_display_name"
      >
        Test!
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        {
          "bottom": 30,
          "left": 20,
          "right": 20,
          "top": 10,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255,255,255,0.08)",
          "borderRadius": 14,
          "height": 28,
          "justifyContent": "center",
          "opacity": 1,
          "width": 28,
        }
      }
      testID="channel_list_header.plus.button"
    >
      <Icon
        name="plus"
        style={
          {
            "color": "rgba(255,255,255,0.8)",
            "fontSize": 18,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingRight": 60,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        {
          "color": "rgba(255,255,255,0.64)",
          "fontFamily": "OpenSans-SemiBold",
          "fontSize": 11,
          "fontWeight": "600",
          "lineHeight": 16,
        }
      }
      testID="channel_list_header.server_display_name"
    />
  </View>
</View>
`;
